src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -308,5 +308,9 @@ private void AddItemToTrimmedList(string assemblyFilePath, string trimmedAssembl
         var trimmedAssemblyItem = new TaskItem(assemblyFilePath);
         trimmedAssemblyItem.SetMetadata("TrimmedAssemblyFileName", trimmedAssemblyFilePath);
         _trimmedAssemblies.Add(trimmedAssemblyItem);
+
+        FileInfo fileInfo = new FileInfo(assemblyFilePath);
```

Comment:
Why is this needed ?

Reply:
The wasm rebuild tests failed, when IL trim is enabled. Test failure log can be found at https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-90436-merge-239a4747ba51427c95/Workloads-Wasm.Build.NativeRebuild.Tests.FlagsChangeRebuildTests/1/console.3814634a.log?helixlogtype=result

It seems that the rebuild mechanism detected that the assembly files are newer than the AOT results. So it tried to re-do AOT. This will hopefully stop that.

Reply:
This sets the timestamp of the trimmed assembly file which is unrelated to AOT.

Reply:
The trimmed assembly files later will replace the original assembly files, which were AOT compiled before being trimmed.

Reply:
I hope (🤞) this won't break anything else. Could you please add a comment here explaining the reasoning?

Reply:
Added comment :)